### ISSUE TITLE
Coredump Crash on King Quest

### ISSUE THREAD

**Issue by gamusin** [iXit/Mesa-3D#272]
I get error message and image distortion as it was some sort of interference, then it crashes. I don't know if it is relevant but when not debugging I get this message:

`nine:nine_state_copy_common_all: Fixed function state not handled properly by StateBlocks.`

I attach a WINEDEBUG= +relay,+xyz  and a coredump.

[https://drive.google.com/file/d/0BxBvyex_4cdlUk5xNmZfVVo5Y2c/view?usp=sharing](url)

[https://drive.google.com/file/d/0BxBvyex_4cdlNTltekVBbTROYW8/view?usp=sharing](url)

**Comment by gamusin** [comment]
trying to debug the coredump nothing really usefull (I do have debuginfo installed btw)
```

warning: core file may not match specified executable file.
[New LWP 10324]
[New LWP 10348]
[New LWP 10349]
[New LWP 10319]
[New LWP 10318]
[New LWP 10272]
[New LWP 10273]
[New LWP 10268]
[New LWP 10347]
[New LWP 10275]
[New LWP 10329]
[New LWP 10278]
[New LWP 10276]
[New LWP 10279]
[New LWP 10277]
[New LWP 10296]
[New LWP 10298]
[New LWP 10301]
[New LWP 10299]
[New LWP 10300]
[New LWP 10297]
[New LWP 10308]
[New LWP 10306]
[New LWP 10307]
[New LWP 10309]
[New LWP 10311]
[New LWP 10310]
[New LWP 10284]
[New LWP 10338]
[New LWP 10313]
[New LWP 10282]
[New LWP 10304]
[New LWP 10305]
[New LWP 10281]
[New LWP 10303]
[New LWP 10302]
[New LWP 10280]
[New LWP 10287]
[New LWP 10312]
[New LWP 10283]
[New LWP 10322]
[New LWP 10330]
[New LWP 10314]
[New LWP 10351]
[New LWP 10350]
Failed to read a valid object file image from memory.
Core was generated by `Z:\guindous\Steam64\steamapps\common\King's Quest\Binaries\Win\KingsQuest.exe '.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0xf0916e30 in ?? ()
[Current thread is 1 (LWP 10324)]
Missing separate debuginfos, use: zypper install wine-32bit-debuginfo-2.1_gallium_nine+staging-2.10.x86_64
(gdb) 

```

**Comment by gamusin** [comment]
strace from kings quest

http://pastebin.com/T5S9uDxh

**Comment by axeldavy** [comment]
could you give a log taken with NINE_DEBUG=all (needs mesa compiled with --enable-debug)

**Comment by gamusin** [comment]
Ok I get a crash window !
`Unhandled exception: page fault on read access to 0x00000001 in 32-bit code (0x01f8a391).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:01f8a391 ESP:0340f11c EBP:0340f13c EFLAGS:00210293(  R- --  I S -A- -C)
 EAX:00000000 EBX:1f855a78 ECX:4302c76d EDX:00000000
 ESI:1d820000 EDI:00000001
Stack dump:
0x0340f11c:  4302c6c1 1f856d2c 1d820000 000060a4
0x0340f12c:  1d820000 0340f19c 021af73f 00000008
0x0340f13c:  0340f148 01f8a89b 1d820000 0340f164
0x0340f14c:  01f89774 00000000 00000000 00000001
0x0340f15c:  00000001 1f856d20 0340f1ac 01f8a8c6
0x0340f16c:  00000001 01f90175 00f61a5d 018a4a4a
Backtrace:
=>0 0x01f8a391 in kingsquest (+0x1b8a391) (0x0340f13c)
  1 0x01f8a89b in kingsquest (+0x1b8a89a) (0x0340f148)
  2 0x01f89774 in kingsquest (+0x1b89773) (0x0340f164)
  3 0x01f8a8c6 in kingsquest (+0x1b8a8c5) (0x0340f1ac)
  4 0x01e9e763 in kingsquest (+0x1a9e762) (0x0340f240)
  5 0x09ddaff6 (0x0340f284)
  6 0x791421db in clr (+0x21da) (0x0340f294)
  7 0x79164a2a in clr (+0x24a29) (0x0340f310)
  8 0x79164bcc in clr (+0x24bcb) (0x0340f450)
  9 0x79164c01 in clr (+0x24c00) (0x0340f46c)
  10 0x79164c21 in clr (+0x24c20) (0x0340f484)
  11 0x7923da9b in clr (+0xfda9a) (0x0340f5e8)
  12 0x7922cf90 in clr (+0xecf8f) (0x0340f850)
  13 0x7922cda4 in clr (+0xecda3) (0x0340fd34)
  14 0x7922d199 in clr (+0xed198) (0x0340fd88)
  15 0x7922d09a in clr (+0xed099) (0x0340fdd4)
  16 0x792aaf00 in clr (+0x16aeff) (0x0340fe0c)
  17 0x603b55ab in mscoreei (+0x55aa) (0x0340fe18)
  18 0x79007f16 in mscoree (+0x7f15) (0x0340fe28)
  19 0x79004de3 in mscoree (+0x4de2) (0x0340fe58)
  20 0x7b4699ec start_process+0xbb(entry=<couldn't compute location>) [/usr/src/debug/wine-2.1/dlls/kernel32/process.c:1133] in kernel32 (0x0340fea8)
  21 0x7bc8b15c call_thread_func_wrapper+0xb() in ntdll (0x0340fed8)
  22 0x7bc8e44d call_thread_func+0x7c(entry=0x7b469930, arg=0x1ff69d2, frame=0x340ffc8) [/usr/src/debug/wine-2.1/dlls/ntdll/signal_i386.c:2815] in ntdll (0x0340ffa8)
  23 0x7bc8b13a RtlRaiseException+0x21() in ntdll (0x0340ffc8)
  24 0x7bc5eb8f start_process+0x3e(kernel_start=0x7b469930) [/usr/src/debug/wine-2.1/dlls/ntdll/loader.c:3461] in ntdll (0x0340ffe8)
  25 0xf75bd50d wine_call_on_stack+0x1c() in libwine.so.1 (0x00000000)
  26 0xf75bd5cb wine_switch_to_stack+0x2a(func=0x7bc5eb50, arg=0x7b469930, stack=0x3410000) [/usr/src/debug/wine-2.1/libs/wine/port.c:77] in libwine.so.1 (0xfff98e58)
  27 0x7bc633ac LdrInitializeThunk+0x29b(kernel_start=<couldn't compute location>, unknown2=<couldn't compute location>, unknown3=<couldn't compute location>, unknown4=<couldn't compute location>) [/usr/src/debug/wine-2.1/dlls/ntdll/loader.c:3523] in ntdll (0xfff98ea8)
  28 0x7b470693 __wine_kernel_init+0x9b2() [/usr/src/debug/wine-2.1/dlls/kernel32/process.c:1327] in kernel32 (0xfff99fc8)
  29 0x7bc6439b __wine_process_init+0x19a() [/usr/src/debug/wine-2.1/dlls/ntdll/loader.c:3732] in ntdll (0xfff9a058)
  30 0xf75bb380 wine_init+0x29f(argc=0x2, argv=0xfff9a5b4, error="", error_size=0x400) [/usr/src/debug/wine-2.1/libs/wine/loader.c:956] in libwine.so.1 (0xfff9a0b8)
  31 0x7c000d9c main+0x8b() in <wine-loader> (0xfff9a518)
  32 0xf73c05d3 __libc_start_main+0xf2() in libc.so.6 (0x00000000)
0x01f8a391: movl	0x0(%edi),%ecx
Modules:
Module	Address			Debug info	Name (232 modules)
PE	  240000-  38b000	Deferred        wxmsw28u_vc_custom
PE	  390000-  3e6000	Deferred        wxmsw28u_aui_vc_custom
PE	  3f0000-  3f7000	Deferred        x3daudio1_7
PE	  400000- 2f40000	Export          kingsquest
PE	 3410000- 34a5000	Deferred        wxmsw28u_xrc_vc_custom
PE	 34b0000- 3575000	Deferred        wxmsw28u_richtext_vc_custom
PE	 3580000- 3636000	Deferred        wxmsw28u_adv_vc_custom
PE	 3640000- 36c0000	Deferred        wxmsw28u_html_vc_custom
PE	 36c0000- 36e2000	Deferred        wxmsw28u_xml_vc_custom
PE	 36f0000- 38f7000	Deferred        d3dcompiler_43
PE	 3900000- 3aff000	Deferred        d3dx9_43
PE	 3b00000- 3b3f000	Deferred        d3dx11_43
PE	 3b40000- 3b54000	Deferred        xapofx1_5
PE	 3b60000- 3b7c000	Deferred        nvtt
PE	 3b80000- 3d0f000	Deferred        libvorbis
PE	 3d10000- 3d19000	Deferred        libogg
PE	 3d20000- 3d2b000	Deferred        libvorbisfile
PE	 3d30000- 3d8c000	Deferred        bink2w32
PE	 3d90000- 418d000	Deferred        physx3checked_x86
PE	 4190000- 419e000	Deferred        nvtoolsext32_1
PE	 41a0000- 4217000	Deferred        physx3cookingchecked_x86
PE	 4220000- 449e000	Deferred        physx3commonchecked_x86
PE	 44a0000- 44b9000	Deferred        easyhook32
PE	 44c0000- 44d9000	Deferred        libresample
PE	 4500000- 4625000	Deferred        gameoverlayrenderer
PE	 f290000- f3cd000	Deferred        steam
PE	10000000-10327000	Deferred        wxmsw28u_core_vc_custom
PE	10800000-1091c000	Deferred        crashhandler
PE	10920000-109ae000	Deferred        xaudio2_7
PE	30000000-302c1000	Deferred        steam2
PE	38000000-38a27000	Deferred        steamclient
PE	3b400000-3b431000	Deferred        steam_api
PE	3f000000-3f117000	Deferred        tier0_s
PE	3f600000-3f656000	Deferred        vstdlib_s
PE	5e0d0000-5e17a000	Deferred        diasymreader
PE	60000000-60021000	Deferred        cserhelper
PE	60340000-6034d000	Deferred        culture
PE	603b0000-60416000	Export          mscoreei
PE	60930000-60940000	Deferred        nlssorting
PE	78050000-780b9000	Deferred        msvcp100
PE	78aa0000-78b5f000	Deferred        msvcr100
PE	79000000-7904a000	Export          mscoree
PE	79060000-7911e000	Deferred        msvcr100_clr0400
PE	79140000-797af000	Export          clr
PE	79810000-79870000	Deferred        clrjit
ELF	7a800000-7a925000	Deferred        opengl32<elf>
  \-PE	7a820000-7a925000	\               opengl32
ELF	7b400000-7b7ef000	Dwarf           kernel32<elf>
  \-PE	7b420000-7b7ef000	\               kernel32
ELF	7bc00000-7bd03000	Dwarf           ntdll<elf>
  \-PE	7bc20000-7bd03000	\               ntdll
ELF	7c000000-7c004000	Dwarf           <wine-loader>
ELF	7cde8000-7ce5d000	Deferred        libpcre.so.1
ELF	7ce60000-7ce87000	Deferred        libselinux.so.1
ELF	7ce88000-7cea0000	Deferred        libresolv.so.2
ELF	7cea0000-7cea5000	Deferred        libkeyutils.so.1
ELF	7cea8000-7cef5000	Deferred        libdbus-1.so.3
ELF	7cef8000-7cf06000	Deferred        libkrb5support.so.0
ELF	7cf08000-7cf3a000	Deferred        libk5crypto.so.3
ELF	7cf40000-7d005000	Deferred        libkrb5.so.3
ELF	7d008000-7d1e6000	Deferred        libcrypto.so.1.0.0
ELF	7d1e8000-7d24c000	Deferred        libssl.so.1.0.0
ELF	7d250000-7d298000	Deferred        libgssapi_krb5.so.2
ELF	7d298000-7d31a000	Deferred        libcups.so.2
ELF	7d358000-7d391000	Deferred        uxtheme<elf>
  \-PE	7d360000-7d391000	\               uxtheme
ELF	7d398000-7d3a4000	Deferred        libxcursor.so.1
ELF	7d3a8000-7d3ad000	Deferred        libcom_err.so.2
ELF	7d3b0000-7d3c2000	Deferred        libavahi-client.so.3
ELF	7d3c8000-7d3d6000	Deferred        libavahi-common.so.3
ELF	7d538000-7d576000	Deferred        libfontconfig.so.1
ELF	7d578000-7d5b4000	Deferred        libpng16.so.16
ELF	7d5b8000-7d5c7000	Deferred        libbz2.so.1
ELF	7d5c8000-7d660000	Deferred        libfreetype.so.6
ELF	7d660000-7d671000	Deferred        libxi.so.6
ELF	7d6b0000-7d6b4000	Deferred        libxinerama.so.1
ELF	7d6c8000-7d6cc000	Deferred        libxcomposite.so.1
ELF	7d6d0000-7d6dc000	Deferred        libxrandr.so.2
ELF	7d6e0000-7d6eb000	Deferred        libxrender.so.1
ELF	7d7f0000-7d884000	Deferred        winex11<elf>
  \-PE	7d800000-7d884000	\               winex11
ELF	7d888000-7d8ae000	Deferred        imm32<elf>
  \-PE	7d890000-7d8ae000	\               imm32
ELF	7d8b0000-7d8c5000	Deferred        faultrep<elf>
  \-PE	7d8c0000-7d8c5000	\               faultrep
ELF	7d8c8000-7d99b000	Deferred        crypt32<elf>
  \-PE	7d8d0000-7d99b000	\               crypt32
ELF	7d9a0000-7d9f1000	Deferred        dsound<elf>
  \-PE	7d9b0000-7d9f1000	\               dsound
ELF	7d9f8000-7da61000	Deferred        dbghelp<elf>
  \-PE	7da00000-7da61000	\               dbghelp
ELF	7da68000-7da93000	Deferred        iphlpapi<elf>
  \-PE	7da70000-7da93000	\               iphlpapi
ELF	7da98000-7db55000	Deferred        msvcrt<elf>
  \-PE	7dab0000-7db55000	\               msvcrt
ELF	7db58000-7db61000	Deferred        librt.so.1
ELF	7db68000-7db71000	Deferred        libffi.so.4
ELF	7db78000-7db8a000	Deferred        libwayland-server.so.0
ELF	7db90000-7db9d000	Deferred        libwayland-client.so.0
ELF	7dba0000-7dbae000	Deferred        libgbm.so.1
ELF	7dbb0000-7dbc2000	Deferred        libdrm.so.2
ELF	7dbc8000-7dbcf000	Deferred        libxxf86vm.so.1
ELF	7dbd0000-7dbd6000	Deferred        libxcb-dri2.so.0
ELF	7dbd8000-7dbf4000	Deferred        libxcb-glx.so.0
ELF	7dbf8000-7dbfe000	Deferred        libxfixes.so.3
ELF	7dc00000-7dc04000	Deferred        libxdamage.so.1
ELF	7dc08000-7dc40000	Deferred        libglapi.so.0
ELF	7dc40000-7dc43000	Deferred        libxshmfence.so.1
ELF	7dc48000-7dc50000	Deferred        libxcb-sync.so.1
ELF	7dc50000-7dc79000	Deferred        libexpat.so.1
ELF	7dc80000-7dc84000	Deferred        libxau.so.6
ELF	7dc88000-7dcbc000	Deferred        libegl.so.1
ELF	7dcc0000-7dd29000	Deferred        libgl.so.1
ELF	7dd30000-7dd43000	Deferred        libxext.so.6
ELF	7dd48000-7de84000	Deferred        libx11.so.6
ELF	7de88000-7de8b000	Deferred        libx11-xcb.so.1
ELF	7de90000-7de99000	Deferred        libxcb-xfixes.so.0
ELF	7dea0000-7dea4000	Deferred        libxcb-present.so.0
ELF	7dea8000-7deac000	Deferred        libxcb-dri3.so.0
ELF	7deb0000-7dec4000	Deferred        psapi<elf>
  \-PE	7dec0000-7dec4000	\               psapi
ELF	7dec8000-7dee4000	Deferred        wsock32<elf>
  \-PE	7ded0000-7dee4000	\               wsock32
ELF	7dee8000-7df1b000	Deferred        d3d9-nine<elf>
  \-PE	7def0000-7def4000	\               d3d9
ELF	7df20000-7df5e000	Deferred        ws2_32<elf>
  \-PE	7df30000-7df5e000	\               ws2_32
ELF	7df60000-7df89000	Deferred        mpr<elf>
  \-PE	7df70000-7df89000	\               mpr
ELF	7df90000-7dfa7000	Deferred        libz.so.1
ELF	7dfb0000-7dfda000	Deferred        libxcb.so.1
ELF	7dfe0000-7e05e000	Deferred        wininet<elf>
  \-PE	7dff0000-7e05e000	\               wininet
ELF	7e060000-7e075000	Deferred        xinput1_3<elf>
  \-PE	7e070000-7e075000	\               xinput1_3
ELF	7e078000-7e094000	Deferred        dinput8<elf>
  \-PE	7e080000-7e094000	\               dinput8
ELF	7e098000-7e153000	Deferred        winmm<elf>
  \-PE	7e0a0000-7e153000	\               winmm
ELF	7e158000-7e2a3000	Deferred        oleaut32<elf>
  \-PE	7e170000-7e2a3000	\               oleaut32
ELF	7e2a8000-7e3f2000	Deferred        ole32<elf>
  \-PE	7e2c0000-7e3f2000	\               ole32
ELF	7e420000-7e464000	Deferred        winspool<elf>
  \-PE	7e430000-7e464000	\               winspool
ELF	7e468000-7e559000	Deferred        comdlg32<elf>
  \-PE	7e470000-7e559000	\               comdlg32
ELF	7e560000-7e5e6000	Deferred        rpcrt4<elf>
  \-PE	7e570000-7e5e6000	\               rpcrt4
ELF	7e5e8000-7e6f7000	Deferred        comctl32<elf>
  \-PE	7e5f0000-7e6f7000	\               comctl32
ELF	7e6f8000-7e713000	Deferred        version<elf>
  \-PE	7e700000-7e713000	\               version
ELF	7e718000-7e84a000	Deferred        gdi32<elf>
  \-PE	7e730000-7e84a000	\               gdi32
ELF	7e850000-7e9b6000	Deferred        user32<elf>
  \-PE	7e860000-7e9b6000	\               user32
ELF	7e9b8000-7ea35000	Deferred        shlwapi<elf>
  \-PE	7e9d0000-7ea35000	\               shlwapi
ELF	7ea38000-7ed0b000	Deferred        shell32<elf>
  \-PE	7ea50000-7ed0b000	\               shell32
ELF	7ed10000-7ed90000	Deferred        advapi32<elf>
  \-PE	7ed20000-7ed90000	\               advapi32
ELF	7ef80000-7efc7000	Deferred        libm.so.6
ELF	e2458000-e24a6000	Deferred        dinput<elf>
  \-PE	e2460000-e24a6000	\               dinput
ELF	ea4b0000-ea4dc000	Deferred        libvorbis.so.0
ELF	ea4e0000-ea4e8000	Deferred        libogg.so.0
ELF	ea4e8000-ea500000	Deferred        libspeex.so.1
ELF	ea500000-ea678000	Deferred        libvorbisenc.so.2
ELF	ea678000-ea6b9000	Deferred        libflac.so.8
ELF	ea6c0000-ea6c6000	Deferred        libgpg-error.so.0
ELF	ea6c8000-ea774000	Deferred        libgcrypt.so.20
ELF	ea778000-ea7a5000	Deferred        liblzma.so.5
ELF	ea7a8000-ea7ae000	Deferred        libcap.so.2
ELF	ea7b0000-ea826000	Deferred        libsndfile.so.1
ELF	ea828000-ea8b9000	Deferred        libsystemd.so.0
ELF	ea8c0000-ea8cc000	Deferred        libjson-c.so.2
ELF	ea8d0000-ea955000	Deferred        libpulsecommon-9.0.so
ELF	ea958000-ea9ac000	Deferred        libpulse.so.0
ELF	ea9e8000-eaa13000	Deferred        winepulse<elf>
  \-PE	ea9f0000-eaa13000	\               winepulse
ELF	eaa18000-eaa3c000	Deferred        mmdevapi<elf>
  \-PE	eaa20000-eaa3c000	\               mmdevapi
ELF	f0428000-f0b9c000	Deferred        d3dadapter9.so.1
ELF	f0ba0000-f0c11000	Deferred        d3d11<elf>
  \-PE	f0bb0000-f0c11000	\               d3d11
ELF	f0d18000-f0d2f000	Deferred        nvapi<elf>
  \-PE	f0d20000-f0d2f000	\               nvapi
ELF	f3a70000-f3a97000	Deferred        libtinfo.so.6
ELF	f3a98000-f3acb000	Deferred        libedit.so.0
ELF	f3c58000-f5e32000	Deferred        libllvm-5.0svn.so
ELF	f5e70000-f6b35000	Deferred        radeonsi_dri.so
ELF	f6ba0000-f6bb8000	Deferred        libelf.so.1
ELF	f6bb8000-f6bc3000	Deferred        libdrm_amdgpu.so.1
ELF	f6bc8000-f6c07000	Deferred        libtxc_dxtn.so
ELF	f6c18000-f6c3d000	Deferred        gameux<elf>
  \-PE	f6c20000-f6c3d000	\               gameux
ELF	f6c40000-f6d83000	Deferred        wined3d<elf>
  \-PE	f6c50000-f6d83000	\               wined3d
ELF	f6d88000-f6db2000	Deferred        dxgi<elf>
  \-PE	f6d90000-f6db2000	\               dxgi
ELF	f6db8000-f6df3000	Deferred        wbemprox<elf>
  \-PE	f6dc0000-f6df3000	\               wbemprox
ELF	f6df8000-f6e14000	Deferred        libgcc_s.so.1
ELF	f6e18000-f6e25000	Deferred        libdrm_radeon.so.1
ELF	f6e28000-f6e31000	Deferred        libdrm_nouveau.so.2
ELF	f6e38000-f6e4e000	Deferred        wlanapi<elf>
  \-PE	f6e40000-f6e4e000	\               wlanapi
ELF	f6f50000-f6f63000	Deferred        gnome-keyring-pkcs11.so
ELF	f6f68000-f6fef000	Deferred        libgmp.so.10
ELF	f6ff0000-f701f000	Deferred        libhogweed.so.2
ELF	f7020000-f7055000	Deferred        libnettle.so.4
ELF	f7058000-f706d000	Deferred        libtasn1.so.6
ELF	f7070000-f70ac000	Deferred        libp11-kit.so.0
ELF	f70b0000-f71c8000	Deferred        libgnutls.so.28
ELF	f71d0000-f7200000	Deferred        p11-kit-trust.so
ELF	f7200000-f7231000	Deferred        netapi32<elf>
  \-PE	f7210000-f7231000	\               netapi32
ELF	f7238000-f726d000	Deferred        secur32<elf>
  \-PE	f7240000-f726d000	\               secur32
ELF	f7270000-f72e5000	Deferred        setupapi<elf>
  \-PE	f7280000-f72e5000	\               setupapi
ELF	f73a0000-f73a5000	Deferred        libdl.so.2
ELF	f73a8000-f7551000	Dwarf           libc.so.6
ELF	f7558000-f7574000	Deferred        libpthread.so.0
ELF	f7580000-f758f000	Deferred        libsensors.so.4
ELF	f7590000-f75ab000	Deferred        imagehlp<elf>
  \-PE	f75a0000-f75ab000	\               imagehlp
ELF	f75b0000-f777f000	Dwarf           libwine.so.1
ELF	f7780000-f77a3000	Deferred        ld-linux.so.2
ELF	f77ab000-f77ac000	Deferred        [vdso].so
Threads:
process  tid      prio (all id:s are in hex)
0000000e services.exe
	[C:\windows\system32\services.exe]
	000000c2    0
	0000003c    0
	00000032    0
	00000028    0
	0000001f    0
	00000012    0
	0000000f    0
00000010 explorer.exe
	[C:\windows\system32\explorer.exe /desktop]
	00000031    0
	00000030    0
	0000002f    0
	0000002e    0
	00000011    0
0000001d winedevice.exe
	[C:\windows\system32\winedevice.exe]
	00000027    0
	00000022    0
	00000021    0
	0000001e    0
00000025 plugplay.exe
	[C:\windows\system32\plugplay.exe]
	0000002b    0
	0000002a    0
	00000026    0
0000002c wdfmgr.exe
	[C:\windows\system32\wdfmgr.exe]
	00000039    0
	00000038    0
	00000037    0
	00000036    0
	00000034    0
	0000002d    0
0000003a winedevice.exe
	[C:\windows\system32\winedevice.exe]
	00000042    0
	0000003f    0
	0000003e    0
	0000003b    0
00000043 Steam.exe
	["C:\Program Files\Steam\steam.exe" "steam://rungameid/345390"]
	000000d2    2
	00000078    0
	00000077    0
	00000076    0
	00000074    0
	00000073    0
	00000072    0
	00000070    0
	0000006f    0
	0000006e    0
	0000006d   15
	0000006c    0
	0000006b    0
	0000006a    0
	00000069    0
	00000068    0
	00000066    1
	00000063    0
	00000061    2
	0000004c    0
	0000004b    0
	0000004a    0
	00000049    0
	00000048    0
	00000045    0
	00000044    0
00000046 steamwebhelper.exe
	["C:\Program Files\Steam\bin\cef\cef.winxp\steamwebhelper.exe" "-cachedir=C:\users\joaco1\Local Settings\Application Data\Steam\htmlcache" "-steampid=67" "-buildid=1484790260" "-steamid=0" --disable-gpu-compositing --disable-gpu --process-per-tab --disable-spell-checking --disable-out-of-process-pac --disable-smooth-scrolling --disable-direct-write]
	00000075    0
	00000060    0
	0000005f    0
	0000005e    0
	0000005d    0
	0000005c    0
	0000005b    0
	0000005a    0
	00000059    0
	00000058    0
	00000057    0
	00000056    0
	00000055    0
	00000054    0
	00000053    0
	00000052    0
	00000051    0
	00000050    0
	0000004f    0
	0000004e    0
	0000004d    0
	00000047    0
000000aa (D) Z:\guindous\Steam64\steamapps\common\King's Quest\Binaries\Win\KingsQuest.exe
	["Z:\guindous\Steam64\steamapps\common\King's Quest\Binaries\Win\KingsQuest.exe"]
	000000e3    0
	000000e2    0
	000000e1    0
	000000e0    0
	000000df    0
	000000d8    0
	000000d1    0
	000000cf    0
	000000cd   15
	000000cc   15
	000000c8   15
	000000c7    0
	000000c6    0
	000000bd    0
	000000b9    0
	000000b8    0
	000000b7    0
	000000b6    0
	000000b5    0
	000000b4    0
	000000b3    0
	000000b2    0
	000000b1    0
	000000b0    0
	000000af    2
	000000ae    0
	000000ab    0 <==
000000d4 GameOverlayUI.exe
	["C:\Program Files\Steam\GameOverlayUI.exe" -pid 170 -manuallyclearframes 0]
	000000de    0
	000000d7    0
	000000d5    0
000000e8 wineconsole.exe
	[wineconsole --use-event=12]
	000000e9    0
System information:
    Wine build: wine-2.1 (Staging)
    Platform: i386
    Version: Windows 5.1 (0)
    Host system: Linux
    Host version: 4.9.12-ck1-5-default
`

and here is the log
[http://pastebin.com/SiZXir7D](url)

**Comment by axeldavy** [comment]
the log did nothing because mesa needs to be compiled with --enable-debug

**Comment by gamusin** [comment]
ok I have to recompile, sorry about that.
I have a snapshot from 11/2 that doesn't have screen corruption , it runs very slow.
I'll post the results in some time.

**Comment by gamusin** [comment]
ok I've recompiled everything but now I cannot reproduce the bug, but it's really really slow maybe 5-10 fps previously it run fine at 50 fps or more.
I'm attaching a link to the nine log.  Mesa git from yesterday.

https://drive.google.com/file/d/0BxBvyex_4cdlZ3h6d0F2OF9wR00/view?usp=sharing

**Comment by gamusin** [comment]
the rendering glitches and errors seams fixed and the crash also as for the performance there is a bug on wine affecting performance on some titles maybe it's that https://bugs.winehq.org/show_bug.cgi?id=42512  (it should be fixed in 2.3)

**Comment by gamusin** [comment]
turned out this was a wine bug closing it's fixed with 2.3
